login-frontend/src/App.tsx

```diff
@@ -55,13 +55,17 @@ export function App() {
           }
         });
 
-        let url = '/api/v1/credentials';
         // get redirect uri query
-        const redirectUri = new URLSearchParams(window.location.search).get('redirect');
-        if (redirectUri) {
-          url = `${url}?redirect=${redirectUri}`;
+        if (!fetchedUri) {
+          const redirectUri = new URLSearchParams(window.location.search).get('redirect');
+          if (redirectUri) {
+            setFetchedUri(`/api/v1/credentials?redirect=${redirectUri}`);
+          } else {
+            console.error('Redirect URI is not present, please go back to original URL');
+            setError(true);
+          }
         }
-        const credentialResponseResponse = await fetch(url, {
+        const credentialResponseResponse = await fetch(fetchedUri, {
```

Comment:
Since you don't have any early return, this will try to send a request to undefined if your state variable is undefined.